template/src/extension.ts

```diff
@@ -1,10 +1,31 @@
-import { Handler, initLSIF, asyncFirst, wrapMaybe } from '../../package/lib'
+import { Handler, initLSIF } from '../../package/lib'
 import * as sourcegraph from 'sourcegraph'
 import { languageSpecs } from '../../languages'
 import { documentSelector } from '../../package/lib/handler'
 
 const DUMMY_CTX = { subscriptions: { add: (_unsubscribable: any) => void 0 } }
 
+// Gets an opaque value that is the same for all locations
+// within a file but different from other files.
+const file = (loc: sourcegraph.Location) =>
+    `${loc.uri.host} ${loc.uri.pathname} ${loc.uri.hash}`
+
+// TODO
+/** circled question mark icons */
+const whiteBadge = `data:image/svg+xml;base64,${btoa(
+    `<svg xmlns='http://www.w3.org/2000/svg' viewBox="0 0 24 24" fill="#ffffff"><path d="M11,18H13V16H11V18M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12C22,6.48 17.52,2 12,2M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,6C9.79,6 8,7.79 8,10H10C10,8.9 10.9,8 12,8C13.1,8 14,8.9 14,10C14,12 11,11.75 11,15H13C13,12.75 16,12.5 16,10C16,7.79 14.21,6 12,6Z"></path></svg>`
+)}`
+const blackBadge = `data:image/svg+xml;base64,${btoa(
+    `<svg xmlns='http://www.w3.org/2000/svg' viewBox="0 0 24 24" fill="#000000"><path d="M11,18H13V16H11V18M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12C22,6.48 17.52,2 12,2M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,6C9.79,6 8,7.79 8,10H10C10,8.9 10.9,8 12,8C13.1,8 14,8.9 14,10C14,12 11,11.75 11,15H13C13,12.75 16,12.5 16,10C16,7.79 14.21,6 12,6Z"></path></svg>`
+)}`
+
+// TODO
+const badge = {
```

Comment:
cc @christinaforney 

Could you provide copy for the hover text and where to link when we show indicators for search-based code intel? That's currently the only indicator we'll have for v0.

Tomorrow I'll show screens of the icons above so that we can change size, shape, color, etc.

Reply:
@efritz - Started doc that this should link to: https://github.com/sourcegraph/sourcegraph/pull/7516

As for hover copy, this could use a bit of improvement but here is a first go:

- For no LSIF data in the repo: This result is provided by search based heuristics. See our documentation for a more precise code intelligence solution.
- For no LSIF data or old data for file and LSIF in repo: This result is provided by search based heuristics because LSIF data does not exist for this file or is out of date.